src/app/ApplicationTemplate.Presentation/ViewModels/Diagnostics/DiagnosticsOverlayViewModel.cs

```diff
@@ -46,19 +46,39 @@ public int ActiveTabIndex
 		await this.GetService<ISectionsNavigator>().OpenModal(ct, () => new DiagnosticsPageViewModel());
 	});
 
+	/// <summary>
+	/// Gets a value indicating whether the diagnostic overlay has been enabled by the user.
+	/// </summary>
 	public bool IsDiagnosticsOverlayEnabled => this.GetFromOptionsMonitor<DiagnosticsOptions, bool>(o => o.IsDiagnosticsOverlayEnabled);
 
-	public bool IsAlignmentGridEnabled
+	/// <summary>
+	/// Gets a value indicating whether gets whether the user has closed manually the diagnostic overlay.
+	/// </summary>
+	public bool IsClosed
 	{
 		get => this.Get<bool>();
-		set => this.Set(value);
+		private set => this.Set(value);
 	}
 
-	public IDynamicCommand ToggleAlignmentGrid => this.GetCommand(() =>
+	public IDynamicCommand CloseDiagnostic => this.GetCommand(() =>
 	{
-		IsAlignmentGridEnabled = !IsAlignmentGridEnabled;
+		IsClosed = true;
 	});
 
+	/// <summary>
+	/// Gets a value indicating whether the diagnostic overlay should be displayed.
+	/// </summary>
+	public bool ShouldDisplayOverlay => this.GetFromObservable(ObserveShouldDisplayOverlay(), initialValue: IsDiagnosticsOverlayEnabled);
```

Comment:
Is this still used or can it be removed?

Reply:
same for ObserveShouldDisplayOverlay

Reply:
Yes it's used. It's the newly added property that the overlay binds its visibility to.